FreeRTOS/Test/CMock/event_groups/event_groups_utest.c

```diff
@@ -0,0 +1,608 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * https://www.FreeRTOS.org
+ * https://github.com/FreeRTOS
+ *
+ */
+/*! @file list_utest.c */
+
+/* C runtime includes. */
+#include <stdlib.h>
+#include <stdbool.h>
+
+/* List includes */
+#include "FreeRTOS.h"
+#include "FreeRTOSConfig.h"
+#include "event_groups.h"
+
+/* Test includes. */
+#include "unity.h"
+#include "unity_memory.h"
+
+/* Mock includes. */
+#include "mock_task.h"
+#include "mock_timers.h"
+#include "mock_list.h"
+#include "mock_list_macros.h"
+#include "mock_fake_assert.h"
+#include "mock_fake_port.h"
+
+
+/* ===========================  DEFINES CONSTANTS  ========================== */
+#define BIT_0            ( 1 << 0 )
+#define BIT_2            ( 1 << 2 )
+#define BIT_4            ( 1 << 4 )
+#define ALL_SYNC_BITS    ( BIT_0 | BIT_2 | BIT_4 )
+
+/* ===========================  GLOBAL VARIABLES  =========================== */
+
+/**
+ * @brief Global message buffer handle used for tests.
+ */
+static EventGroupHandle_t xEventGroupHandle;
+static List_t xListTemp = { 0 };
+static List_t * pxListTemp = &xListTemp;
+static ListItem_t xListItemDummy = { 0 };
+static ListItem_t * pxListItem_HasTaskBlockOnBit0 = &xListItemDummy;
+
+/* ===========================  EXTERN VARIABLES  =========================== */
+
+/* ==========================  CALLBACK FUNCTIONS =========================== */
+
+void * pvPortMalloc( size_t xSize )
+{
+    return unity_malloc( xSize );
+}
+void vPortFree( void * pv )
+{
+    return unity_free( pv );
+}
+
+/* ============================  Unity Fixtures  ============================ */
+/*! called before each testcase */
+void setUp( void )
+{
+    xEventGroupHandle = NULL;
+
+    pxListTemp->pxIndex = ( ListItem_t * ) &( pxListTemp->xListEnd );
+    pxListTemp->xListEnd.xItemValue = portMAX_DELAY;
+    pxListTemp->xListEnd.pxNext = ( ListItem_t * ) &( pxListTemp->xListEnd );
+    pxListTemp->xListEnd.pxPrevious = ( ListItem_t * ) &( pxListTemp->xListEnd );
+    pxListTemp->uxNumberOfItems = ( UBaseType_t ) 0U;
+
+    pxListItem_HasTaskBlockOnBit0->xItemValue = BIT_0;
+
+    vFakeAssert_Ignore();
+    vFakePortEnterCriticalSection_Ignore();
+    vFakePortExitCriticalSection_Ignore();
+    ulFakePortSetInterruptMaskFromISR_IgnoreAndReturn( 0U );
+    vFakePortClearInterruptMaskFromISR_Ignore();
+
+    /* Track calls to malloc / free */
+    UnityMalloc_StartTest();
+}
+
+/*! called after each testcase */
+void tearDown( void )
+{
```

Comment:
Please clean up your mocks between test cases so that a single test case failure does not cascade down to the next test case.
For each mocked file, call it's _Verify() and _Destroy() functions here.
mock_list_Verify();
mock_list_Destroy();

Reply:
I didn't see this from the Cmock doc. So I checked the mock runner code, it looks it is called when running each test cases. So we don't have to call it.